documentatie/flowcharts/data-entry-forms.md

```diff
@@ -0,0 +1,219 @@
+    error-cur-page{"error for 
+        current page?"}
+    warning-cur-page{"warning for
+        current page?"}
+    warnings-accepted{"warning(s) accepted?"}
+    user-addresses-error{user addresses error}
+
+    user-addresses-warning{user addresses warning}
+
+    click-next(click next)
+    call-api(call api)
+    change-input(change input)
+    accept-warning(accept warning)
+
+    go-to-next-page([go to next page])
+
+    %% flow
+    flow-start --> click-next
+    click-next --> call-api
+    call-api --> error-any-prev-page
+
+    error-any-prev-page -- yes --> go-to-prev-page
+
+    error-any-prev-page -- no --> error-cur-page
+
+    error-cur-page -- yes --> user-addresses-error
+    
+    user-addresses-error -- abort --> abort-input
+    user-addresses-error -- resolve --> change-input
+    change-input --> click-next
+
+    error-cur-page -- no --> warning-cur-page
+    warning-cur-page -- yes --> warnings-accepted
+    warnings-accepted -- no --> user-addresses-warning
+    warnings-accepted -- yes --> go-to-next-page
+    user-addresses-warning -- resolve --> change-input
+    user-addresses-warning -- accept --> accept-warning
+    accept-warning --> click-next
+    user-addresses-warning -- abort --> abort-input
+
+    warning-cur-page -- no --> go-to-next-page
+
+```
+
+Note that if there is a warning and the user changes the input, they should no longer have the option to accept the warning. They need to click "Next" first, to validate the changed input.
+
+
+## Click navigation item or browser back/forward buttons
+
+```mermaid
+flowchart TD
+    %% start and end
+    flow-start([start])
+
+    go-to-page([go to selected page])
+    remain-on-page([remain on current page])
+
+    %% steps
+    click-nav-item(click navigation item)
+    click-browser-back("click browser
+        back button")
+    click-browser-forward("click browser
+        forward button")
+    on-furthest-page{on furthest page?}
+    user-made-changes{user made changes?}
+    save-changes{save changes?}
+    call-api(call api)
+    cache-input(cache input)
+    reset-changes(reset changes)
+    errors-or-warnings{errors or warnings?}
+
+    %% flow
+    flow-start --> click-nav-item
+    flow-start --> click-browser-back
+    flow-start --> click-browser-forward
+    click-browser-back --> on-furthest-page
+    %% if you can click the browser forward button, you by definition are not on the furthest page
+    click-browser-forward --> user-made-changes
+    click-nav-item --> on-furthest-page
+    on-furthest-page -- yes --> cache-input
+    cache-input --> go-to-page
+
+    on-furthest-page -- no --> user-made-changes
+    user-made-changes -- no --> go-to-page
+    user-made-changes -- yes --> save-changes
+    save-changes -- no --> reset-changes
+    reset-changes --> go-to-page
+    save-changes -- yes --> call-api
+    call-api --> errors-or-warnings
+    errors-or-warnings -- no --> go-to-page
+    errors-or-warnings -- yes --> remain-on-page
```

Comment:
wordt de error of warning hier ook gelijk weergegeven of is dat een aparte stap?

Reply:
Die worden weergegeven. Alle logica daarvoor staat beschreven in de "Render form" flowchart.